packages/cli/lib/lib/webpack/webpack-client-config.js

```diff
@@ -226,61 +225,36 @@ function isProd(config) {
 		},
 	};
 
-	if (config.esm) {
+	if (config.esm && config.sw) {
 		prodConfig.plugins.push(
-			new BabelEsmPlugin({
-				filename: '[name].[chunkhash:5].esm.js',
-				chunkFilename: '[name].chunk.[chunkhash:5].esm.js',
-				excludedPlugins: ['BabelEsmPlugin', 'SWBuilderPlugin'],
-				beforeStartExecution: (plugins, newConfig) => {
-					const babelPlugins = newConfig.plugins;
-					newConfig.plugins = babelPlugins.filter(plugin => {
-						if (
-							Array.isArray(plugin) &&
-							plugin[0].indexOf('fast-async') !== -1
-						) {
-							return false;
-						}
-						return true;
-					});
-					plugins.forEach(plugin => {
-						if (
-							plugin.constructor.name === 'DefinePlugin' &&
-							plugin.definitions
-						) {
-							for (const definition in plugin.definitions) {
-								if (definition === 'process.env.ES_BUILD') {
-									plugin.definitions[definition] = true;
-								}
-							}
-						} else if (
-							plugin.constructor.name === 'DefinePlugin' &&
-							!plugin.definitions
-						) {
-							throw new Error(
-								'WebpackDefinePlugin found but not `process.env.ES_BUILD`.'
-							);
-						}
-					});
-				},
+			new InjectManifest({
+				swSrc: fallbackSwPath,
+				swDest: 'sw-esm.js',
+				include: [
+					/^\/?index\.html$/,
+					/\.esm.js$/,
+					/\.css$/,
+					/\.(png|jpg|svg|gif|webp)$/,
+				],
+				webpackCompilationPlugins: [
+					new webpack.DefinePlugin({
+						'process.env.ESM': true,
+					}),
+				],
 			})
 		);
-		config.sw &&
-			prodConfig.plugins.push(
-				new InjectManifest({
-					swSrc: 'sw-esm.js',
-					include: [/^\/?index\.html$/, /\.esm.js$/, /\.css$/, /\.(png|jpg)$/],
-					precacheManifestFilename: 'precache-manifest.[manifestHash].esm.js',
-				})
-			);
 	}
 
 	if (config.sw) {
```

Comment:
Is this going to add a second `InjectManifest` plugin when `config.esm` is `true`?

Reply:
yes that is for the second service worker.